content/docs/rfcs/72/README.md

```diff
@@ -0,0 +1,297 @@
+The keystore COULD use PBKDF2 password-based encryption, 
+as described in [RFC 2898](https://www.ietf.org/rfc/rfc2898.txt).
+
+A `WakuCredential` object MUST include:
+| Name | Description  |
+|----|-----|
+| password | used to encrypt keystore and decryption key |
+| secret | key to be encrypted |
+| pubKey | public key |
+| path | HD, hardened derivation, path used to generate the secret |
+| checksum | hashing function | 
+| cipher | cipher function |
+
+```js
+
+crypto: {
+	
+	cipher: "string" // The cipher function
+	cipherparams: {
+		iv: "string" // The cipher parameters
+	},
+	ciphertext: "string" // The cipher message,
+	kdf: "string" // KDF Function,
+	kdfparams: {
+		param: integer // Salt value and iteration count,
+		dklen: integer // Length in octets of derived key, MUST be positive integer,
+		c: "string" // Iteration count, MUST be positive integer,
+		prf: "string" // Underlying pseudorandom function,
+		salt: "string" // Produces a large set of keys based on the password
+	},
+	mac: "string" // Checksum
+}
+
+```
+	
+### Decryption
+The keystore SOULD decrypt a user's credentials using a password and 
+a Merkle proof with PBKDF2 than returns the secret key.
+- To generate the `decryptionKey`, it MUST be constructed from a password and KDF.
+- The `decryptionKey`, cipher function and cipher parameters MUST be used to encrypt the secret key.
+The cipher function encrypts the secret key using the `decryptionKey`.
+- If the `decryptionKey` is longer than the key size required by the cipher,
+it MUST be truncated to the correct number of bits.
+
+## Test Vectors
+### Input:
+Hashing function used: Poseidon Hash, as described in [Poseidon Paper](https://eprint.iacr.org/2019/458.pdf)
+
+`application`: "waku-rln-relay"
+
+`appIdentifier`: "01234567890abcdef"
+
+`version`: "0.2" 
+
+`hash_function`: "poseidonHash"
+
+`password`: "sup3rsecure"
+
+```js
+identityCredential = {
+	IDTrapdoor: [
+		211, 23, 66, 42, 179, 130, 131, 111, 201, 205, 244, 34, 27, 238, 244,
+        	216, 131, 240, 188, 45, 193, 172, 4, 168, 225, 225, 43, 197, 114, 176,
+        	126, 9,
+	],
+	IDNullifier: [
+        	238, 168, 239, 65, 73, 63, 105, 19, 132, 62, 213, 205, 191, 255, 209, 9,
+        	178, 155, 239, 201, 131, 125, 233, 136, 246, 217, 9, 237, 55, 89, 81,
+        	42,
+	],
+	IDSecretHash: [
+        	150, 54, 194, 28, 18, 216, 138, 253, 95, 139, 120, 109, 98, 129, 146,
+        	101, 41, 194, 36, 36, 96, 152, 152, 89, 151, 160, 118, 15, 222, 124,
+        	187, 4,
+	],
+	IDCommitment: [
+        	112, 216, 27, 89, 188, 135, 203, 19, 168, 211, 117, 13, 231, 135, 229,
+		58, 94, 20, 246, 8, 33, 65, 238, 37, 112, 97, 65, 241, 255, 93, 171, 15,
+	],
+	IDCommitmentBigInt: buildBigIntFromUint8Array(
+        	new Uint8Array([
+          		112, 216, 27, 89, 188, 135, 203, 19, 168, 211, 117, 13, 231, 135, 229,
+          		58, 94, 20, 246, 8, 33, 65, 238, 37, 112, 97, 65, 241, 255, 93, 171,
+          		15,
+		])
+	)
```

Comment:
```suggestion

```

Reply:
Does the `identityCredential` belong in the decryption section? I recently moved it from the input section. @rymnc

Reply:
its part of the input actually, it isn't used to decrypt the keystore.

Reply:
@rymc Yes, I see now, so I forgot to add `contractId` to the `membershipHash` creation, which made me a little confused. I changed it to include the four attributes, instead of three previously, [here](https://github.com/jimstir/rfc/blob/521967fb7ca20853e7ff0f04ca1920dfe74e7a80/content/docs/rfcs/72/README.md?plain=1#L97) and [here](https://github.com/jimstir/rfc/blob/521967fb7ca20853e7ff0f04ca1920dfe74e7a80/content/docs/rfcs/72/README.md?plain=1#L114)